Decrease minimum glibc version

In https://github.com/conda-forge/gtk4-feedstock/commit/07c207502f8ae43d470c5ec1320bd565e24d7ea6, the minimum glibc version was bumped to 2.34. At a first glace, I'm unsure why the bump was needed, as most dependencies still have the minimum version set to 2.17. Would it be possible to relax this requirement to either glibc 2.17 or 2.28?

The reason for this request is that I've recently started looking into using conda-forge packages for building an AppImage, and a minimum glibc version of 2.28 or below would allow it to run on some distros like Rocky Linux 8 and Slackware 15.

Looks like the aarch64 build fails with glibc 2.17 due to:

```
../gdk/gdkdmabuf.c: In function 'gdk_dmabuf_new_for_bytes':
../gdk/gdkdmabuf.c:597:22: error: 'F_ADD_SEALS' undeclared (first use in this function)
  597 |   if (fcntl (mem_fd, F_ADD_SEALS, F_SEAL_SHRINK) < 0)
      |                      ^~~~~~~~~~~
../gdk/gdkdmabuf.c:597:22: note: each undeclared identifier is reported only once for each function it appears in
../gdk/gdkdmabuf.c:597:35: error: 'F_SEAL_SHRINK' undeclared (first use in this function)
  597 |   if (fcntl (mem_fd, F_ADD_SEALS, F_SEAL_SHRINK) < 0)
      |                                   ^~~~~~~~~~~~~

```

glibc 2.28 works though. I'll open a PR lowering the version requirement to 2.28.

--- a/.ci_support/linux_64_.yaml
+++ b/.ci_support/linux_64_.yaml
@@ -5,7 +5,7 @@ c_compiler_version:
 c_stdlib:
 - sysroot
 c_stdlib_version:
-- '2.34'
+- '2.28'
 cairo:
 - '1'
 channel_sources:

--- a/.ci_support/linux_aarch64_.yaml
+++ b/.ci_support/linux_aarch64_.yaml
@@ -5,7 +5,7 @@ c_compiler_version:
 c_stdlib:
 - sysroot
 c_stdlib_version:
-- '2.34'
+- '2.28'
 cairo:
 - '1'
 channel_sources:

--- a/.ci_support/linux_ppc64le_.yaml
+++ b/.ci_support/linux_ppc64le_.yaml
@@ -5,7 +5,7 @@ c_compiler_version:
 c_stdlib:
 - sysroot
 c_stdlib_version:
-- '2.34'
+- '2.28'
 cairo:
 - '1'
 channel_sources:

--- a/recipe/conda_build_config.yaml
+++ b/recipe/conda_build_config.yaml
@@ -1,2 +1,2 @@
 c_stdlib_version:
-  - "2.34"                 # [linux]
+  - "2.28"                 # [linux]

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -13,7 +13,7 @@ source:
     - patches/001-win32-fix-black-borders.patch
 
 build:
-  number: 0
+  number: 1
   run_exports:
     - {{ pin_subpackage('gtk4', max_pin='x') }}
     # gtk apps need at least a default icon set for fallback

